kroxylicious-app/src/test/java/io/kroxylicious/app/KroxyliciousIT.java

```diff
@@ -43,11 +51,76 @@ class KroxyliciousIT {
     private static final String TOPIC_1 = "my-test-topic";
     private static final String TOPIC_2 = "other-test-topic";
     private static final String PLAINTEXT = "Hello, world!";
+    private static final ProxyEnvironment DEFAULT_ENV = ProxyEnvironment.PRODUCTION;
 
     @Test
-    void shouldProxyWhenRunAsStandaloneProcess(KafkaCluster cluster, Admin admin, @TempDir Path tempDir) throws Exception {
-        var proxyAddress = HostPort.parse("localhost:9192");
+    void shouldFailToStartWithInternalConfigurationsByDefault(@TempDir Path tempDir) throws IOException {
+        SubprocessKroxyliciousFactory kroxyliciousFactory = new SubprocessKroxyliciousFactory(tempDir, (environment, processBuilder) -> {
+            // no-op so that io is not inherited
+        }, List.of());
+        var tester = kroxyliciousTester(proxy("fake:9092").withInternal(Map.of("a", "b")), kroxyliciousFactory);
+        Process lastProcess = kroxyliciousFactory.lastProcess;
+        assertThat(lastProcess).isNotNull();
+        assertThat(lastProcess.onExit()).succeedsWithin(5, TimeUnit.SECONDS);
+        byte[] bytes = lastProcess.getInputStream().readAllBytes();
+        String output = new String(bytes, StandardCharsets.UTF_8);
+        assertThat(output).contains("internal configuration for proxy present in production environment");
```

Comment:
To avoid introducing the term "internal" we could just say "development mode configurations present without developer mode enabled"